gen/definitions/network_devices.yaml

```diff
@@ -0,0 +1,35 @@
+---
+name: Network Devices
+rest_endpoint: /dna/intent/api/v1/network-device
+get_no_id: true
+no_data_source: true
```

Comment:
I think we could still auto-generate the data source code and just replace specific sections of the generated code. Each rendered .go file has a number of sections (`//template:begin` and `//template:end`), where if you want to change a certain section you would make the necessary changes, then remove the begin/end template markers, which will prevent this specific section of code from being overwritten the next time you would run `go generate`.

Reply:
In this particular place I had additional problem, which I couldn't solve with deleting `//template:begin` and `//template:end`. By default, the generated `docs` and `examples` of data source always tells users to use `id` attribute:

```hcl2
data "catalystcenter_network_devices" "example" {
  id = "76d24097-41c4-4558-a4d0-a8c07ac08470"
}
```

But in this particular case, the `id` doesn't fit well. It is confusing to users to even show them an `id` attribute because this source always returns all devices, whereas any user would probably confuse that `id` with the device's UUID.

The `no_data_source` is probably the least invasive way to prevent gen/generator.go from adding `id` back to the `docs` and `examples`.

So now the only remaining contribution from gen/generator.go is partial templating of model.go, and I agree it will be useful for any future attribute additions.

Reply:
Acknowledged. The phrase "singleton data source" brings back a bit of GoF memories :)